processor/metricstarttimeprocessor/internal/subtractinitial/adjuster.go

```diff
@@ -127,7 +129,7 @@ func adjustMetricHistogram(referenceTsm, previousValueTsm *datapointstorage.Time
 
 		if datapointstorage.IsResetHistogram(currentDist, previousTsi.Histogram) {
 			// reset re-initialize everything and use the non adjusted points start time.
-			resetStartTimeStamp := pcommon.NewTimestampFromTime(pointStartTime.AsTime().Add(-1 * time.Millisecond))
+			resetStartTimeStamp := pcommon.NewTimestampFromTime(currentDist.Timestamp().AsTime().Add(-1 * time.Millisecond))
```

Comment:
This is needed because the start timestamp can be missing (which is the whole point of the processor).  Setting the start time to (0 - 1 ms) rolls over to close to max int64, which is definitely not what we want.  I think this was just a mistake/typo that was copied over from the prometheus receiver.